### PR TITLE
Resolves #1405

### LINKED ISSUE #1405: Implement shortlinks in all code references
## Issue Overview

We've implemented shortlinks visibly in the plugin, but there are many links to articles in the code itself, specifically for translators and other miscellaneous purposes. Let's get ALL of them replaced correctly with shortlinks.

### PR DESCRIPTION
**Pull request by mathetos**
Fixes #1405 

Updates all internal links to documentation to shortlinks.

### CODE REVIEW

File: includes/admin/settings/class-settings-display.php [hunk 1]
```diff
@@ -65,8 +65,8 @@ public function get_settings() {
 						),
 						array(
 							'name' => esc_html__( 'Floating Labels', 'give' ),
-							/* translators: %s: https://givewp.com/documentation/core/give-forms/creating-give-forms/#floating-labels */
-							'desc' => sprintf( wp_kses( __( '<a href="%s" target="_blank">Floating labels</a> in set form labels within fields and can improve the donor experience. Note that if the "Disable CSS" option is enabled, you will need to style the floating labels yourself.', 'give' ), array( 'a' => array( 'href' => array(), 'target' => array() ) ) ), esc_url( 'https://givewp.com/documentation/core/give-forms/creating-give-forms/#floating-labels' ) ),
+							/* translators: %s: http://docs.givewp.com/floatinglabels */
```
Comment (DevinWalker) [inline includes/admin/settings/class-settings-display.php]: Can you adjust this to `floating-labels` to match other references to `floating-labels`?

File: includes/admin/class-give-settings.php [hunk 2]
```diff
@@ -1414,10 +1414,10 @@ function give_api_callback() {
 	?>
 	<span class="cmb2-metabox-description api-description">
 		<?php echo sprintf(
-		/* translators: 1: https://givewp.com/documentation/give-api-reference/ 2: https://givewp.com/addons/zapier/ */
+		/* translators: 1: http://docs.givewp.com/api 2: http://docs.givewp.com/addon-zapier */
 			__( 'You can create API keys for individual users within their profile edit screen. API keys allow users to use the <a href="%1$s" target="_blank">Give REST API</a> to retrieve donation data in JSON or XML for external applications or devices, such as <a href="%2$s" target="_blank">Zapier</a>.', 'give' ),
-			esc_url( 'https://givewp.com/documentation/give-api-reference/' ),
-			esc_url( 'https://givewp.com/addons/zapier/' )
+			esc_url( 'http://docs.givewp.com/api' ),
```
Comment (DevinWalker) [inline includes/admin/class-give-settings.php]: should this link be more specific that `api`?

File: includes/admin/settings/class-settings-display.php (hunk 1 above)
Comment (mathetos) [inline includes/admin/settings/class-settings-display.php]: Sorry, for some reason some of my commits didn't get pushed. Should be resolved with this latest commit.

File: includes/admin/class-give-settings.php (hunk 2 above)
Comment (mathetos) [inline includes/admin/class-give-settings.php]: I'd love to write more docs about the API, so this one would be the root doc, and additional docs would be sub-pages and therefore need a hyphen, like `api-extending` That's why I kept it short.

**Pull request merged**